docs/modules/ROOT/pages/advanced-how-tos/managing-multiple-versions.adoc

```diff
@@ -1,25 +1,24 @@
 * A *ProjectDevelopmentStream* indicates an independent stream of development. A *ProjectDevelopmentStream* can contain one or more *Applications* each containing one or more *Components*.
-* As described above, starting a new development stream involves creating a large  amount of  *Application* and *Component* resources. The Project Controller helps to streamline that by allowing to create a *ProjectDevelopmentStreamTemplate* resource that specifies the resources to be created and allows for using variables to customize them. Using a template, many similar development streams can be created quickly.
+* As described above, starting a new development stream involves creating a large  amount of  *Application*, *Component* and other resources. The Project Controller helps to streamline that by allowing to create a *ProjectDevelopmentStreamTemplate* resource that specifies the resources to be created and allows for using variables to customize them. Using a template, many similar development streams can be created quickly.
 
 == Using the Project Controller
 
-In the sections below we will describe how to use the custom resources supported by the Project Controller to manage multiple software versions and generate Application and Component resources.
+In the sections below we will describe how to use the custom resources supported by the Project Controller to manage multiple software versions and generate Application, Component and other resources.
 
 === Before you begin
 
-It's possible to have the Project Controller create all the Application and Component resources you would need. It is recommended, however, for the first time a particular repository is on-boarded, to do so manually via the {ProductName} UI. That would help ensuring the right access credentials are in place as well as that the Tekton Pipeline-as-Code files are committed to the repo.
+The typical process of onboarding a componnent to {ProductName} includes having the {ProductName} Build Service generate a PR to the component Git repository which includes the Tekton Pipeline-as-Code (PaC) pipeline definition for building that component. As a part of this process various other resources such as the PaC Repository resource and Secret resources for accessing the repository are also created.
 
-When creating components the Project Controller assumes that:
+Whe using the Project Controller to create components the process of generating the PR is skipped because it is assumed the branches being created in the component Git repositores already contain PaC pipeline files. Eert Git repository, however, needs to undergo this process at least once, therefore it is recommended to onboard at least once component from each repository from the UI or by other means before attempting to manage components in that repository via the Project Controller.
```

Comment:
```suggestion
When using the Project Controller to create components the process of generating the PR is skipped because the build service assumes the branches being created in the component Git repositories already contain PaC pipeline files. Each Git repository, however, needs to undergo this process at least once, therefore it is recommended to onboard at least once component from each repository from the UI or by other means before attempting to manage components in that repository via the Project Controller.
```
I think that this wording is confusing though. Would it be better to say something like this?
```suggestion
When new components are generated with the Project Controller, PRs adding the Tekton PipelineRuns needed by PAC are not created. In order to properly trigger pipelines on push and pull request events to the repository, these definitions need to be added my some other means. It is recommended that one branch is onboarded to {ProductName} without the Project Controller to get these files pushed. These definitions can then be copied to other branches that are onboarded to the Project Controller.
```
Will there be a PR to rectify the component names for the branches that have the Tekton files copied from another branch (with another component name)? Or will the PipelineRun be updated with new triggers for the branch, or will users need to perform this manually as well?

Reply:
The inmortant part of the "PR" creation process the build service does is IMO the creationg of the PaC Repository resource and accompanying secret. Your 2nd suggestion hoever seems to place the empasys on the PaC files which could be obtained by other means. Also the language seems to imply that users would need to copy PaC files between branches. That would not IMO be the typical case. The typical case would be to onboard the `main` branch manually and then create the `v1.0.0` from it. That way, the `v1.0.0` would already containg PaC files (Albait needing sume adjustments). Given that I decided to go with your first suggestion.

For your 2nd question, there is no automated mechanism to create piplien adjustment PRs for the users, and no such mechanism is planned. We have instructions below about how to manually update the pipeline files as needed.

I suppose that with some component naming conventions (E.g. `comp-name-{{branchName}}`) we could come up with a multi-branch PipelineRun files that would not need adjustments and could also possibly render the current project-controller unneccessary (By, e.g creating the component resource it is building if its missing), but that would be out of scope for the current work.